examples/workflows/script_artifact_passing.py

```diff
@@ -0,0 +1,23 @@
+from hera.workflows import Artifact, Steps, Workflow, script
+
+
+@script(outputs=Artifact(name="hello-art", path="/tmp/hello_world.txt"))
+def whalesay():
+    with open("/tmp/hello_world.txt", "wb") as f:
+        f.write("hello world")
+
+
+@script(inputs=Artifact(name="message", path="/tmp/message"))
+def print_message():
+    with open("//tmp/message", "rb") as f:
+        message = f.readline()
+    print(message)
+
+
+with Workflow(generate_name="artifact-passing-", entrypoint="artifact-example") as w:
+    with Steps(name="artifact-example") as s:
+        whalesay(name="generate-artifact")
+        print_message(
+            name="consume-artifact",
+            arguments=[Artifact(name="message", from_="{{steps.generate-artifact.outputs.artifacts.hello-art}}")],
```

Comment:
Is this PR a nice opportunity to implement `get_artifact` on step/task? 🤔 The same way there is a `get_parameter`

Reply:
I want to make a new PR where I'll do all the tasks in #595, and then the diff will show the changes from argo expressions to native hera 😄